### ISSUE TITLE
Transpiling of yarn workspaces broke after upgrading to next 9.4.0

### ISSUE THREAD

**Issue by tbergq** [martpie/next-transpile-modules#85]
- [x] I HAVE READ [THE FAQ](https://github.com/martpie/next-transpile-modules#faq) AND MY PROBLEM WAS NOT DESCRIBED THERE

**Are you trying to transpile a local package or an npm package?**
local

**Describe the bug**
Transpiling worked with next `9.3.6`, but broke after upgrading to `9.4.0`.

```
/Users/trondbergquist/code/adeira/universe/node_modules/@adeira/relay/src/index.js
SyntaxError: /Users/trondbergquist/code/adeira/universe/node_modules/@adeira/relay/src/index.js: Support for the experimental syntax 'flow' isn't currently enabled (53:8):

  51 | };
  52 | 
> 53 | export type { RelayProp } from './createFragmentContainer';
     |        ^
  54 | export type { PaginationRelayProp } from './createPaginationContainer';
  55 | export type { RefetchRelayProp } from './createRefetchContainer';
  56 | export type { DeclarativeMutationConfig } from './types.flow';
```

**To Reproduce**
Upgrade from next 9.3.6 to 9.4.0 and transpiling of yarn workspaces breaks

**Expected behavior**
It should still be working

**Setup**

- Next.js version: 9.4.0
- `next-transpile-modules` version: 3.3.0
- Node.js version: 14.2.0
- `npm`/`yarn` version: yarn 1.22.4
- Operating System: OS X

**Additional context**

**Comment by TrySound** [comment]
Show your next.config.js please

**Comment by tbergq** [comment]
next.config.js file can be seen [here](https://github.com/adeira/universe/blob/master/src/example-relay/next.config.js)

**Comment by martpie** [comment]
Can you give more information about your setup? Do you own the transpiled files? Are you using flow? Why are flow files transpiled? Why are you transpiling  an example repository? etc.

I am lacking some context on if this is the responsibility of `next-transpile-modules` or `next`

**Comment by tbergq** [comment]
> Can you give more information about your setup?

We have a monorepo using yarn workspaces. We distribute several npm packages, among them some wrappers around relay. The relay example repository is there to show some examples of how to use relay. Mutations, paginations etc. 

> Do you own the transpiled files?

Yes

> Are you using flow?

Yes

> Why are flow files transpiled?

Because flow annotation is not valid javascript syntax, the flowtypes have to be stripped.

> Why are you transpiling an example repository?

Same as above I would say. It needs to be transpiled to run in the browser. 

At the end of the day, our setup worked with next `9.3.6`, and it broke when we upgrade to `9.4.0`. 

Maybe it is not the responsibility of this package, but it is clear from the error message that our workspace is no longer transpiled, so that is why I started here.

**Comment by tbergq** [comment]
Actually, I just discovered that doing `yarn next build && yarn next start` works, while `yarn next` produces the error 🤔 

So the error might not be in `next-transpile-modules`. It seems to be correctly transpiled for production.

**Comment by martpie** [comment]
could be interesting to check the changelogs of next, they may have modified the base Babel config, or there may have been a Webpack/Babel update.

Last: do you have a custom Babel config? If yes, can you post it here?

**Comment by tbergq** [comment]
Yes, we have this babel config

```

module.exports = {
  presets: ['@adeira/babel-preset-adeira', 'next/babel'],
  plugins: ['relay'],
};
```

**Comment by tbergq** [comment]
> could be interesting to check the changelogs of next, they may have modified the base Babel config, or there may have been a Webpack/Babel update.


From reading the blog about the `9.4.0` release, the only thing I can think about is that somehow related, is the `Import alias` functionality they added. Could it be that it interprets our workspace as an import alias 🤔

**Comment by martpie** [comment]
You don't have something for flow in your Babel config?

**Comment by tbergq** [comment]
Yes, we do. This project has existed for about 3 years. There are no problems with Babel or other setup.

Something changed in next 9.4.0, and I suspected that something would need to change with next-transpile-modules also.

**Comment by andrewmclagan** [comment]
We "may" also be experiencing this issue. Just bumped to `9.4.x` and we are seeing issues around the new optional chaining language feature. We were using this prior to the upgrade, not code has changed. Our assumption is transpiration has stopped for some reason?

Also when we do `next build` it works just `next dev` 

Side note we are using Yarn v2

**Comment by martpie** [comment]
@andrewmclagan Can you share your Next config as well please?

**Comment by andrewmclagan** [comment]
```javascript
const path = require('path');
const withPlugins = require('next-compose-plugins');
const transpileModules = require('next-transpile-modules');

const withModules = transpileModules([
  '@pkg/mui-theme',
  '@pkg/forms',
  '@pkg/sdk',
  '@pkg/client',
  '@pkg/graphql'
]);

const nextConfig = {
  webpack: config => {
    config.resolve.alias['components'] = path.join(__dirname, 'src/components');
    config.resolve.alias['dialogs'] = path.join(__dirname, 'src/dialogs');
    config.resolve.alias['layouts'] = path.join(__dirname, 'src/layouts');
    config.resolve.alias['forms'] = path.join(__dirname, 'src/forms');
    config.resolve.alias['lib'] = path.join(__dirname, 'src/lib');
    return config;
  }
};

module.exports = withPlugins([withModules], nextConfig);
```

**Comment by andrewmclagan** [comment]
If we keep all the above versions locked and simply use yarn `1.22` things work. So it seems Yarn 2 is not working with this package, it works with a vanilla next app. Also yarn `1.22` plug-and-play is not working.

**Comment by tbergq** [comment]
For me this is not related to `Yarn PnP` since I am already using yarn `1.22.4`. 

I made a small reproduction repo. You can check it out [here](https://github.com/tbergq/next-transpile-error). The master branch is using `9.4.x`, and the error is reproducable. There is a branch called `9.3.6` that is working with the same setup.

**Comment by martpie** [comment]
It looks a lot for me like a Babel misconfiguration. I will try to invest some time today or tomorrow to check that.

Thanks for the repro, it helps a lot!

**Comment by tbergq** [comment]
> It looks a lot for me like a Babel misconfiguration

I would buy this if it never worked, the thing that makes me think this is not it, is that it broke after upgrading. 

> I will try to invest some time today or tomorrow to check that.

Thanks, I appreciate that a lot 🙂

**Comment by martpie** [comment]
> I would buy this if it never worked, the thing that makes me think this is not it, is that it broke after upgrading.

I buy it because the Next.js team always tweaks the internal and Webpack and Babel configuration in each release. We had breaking changes in the past because I don't read all the Next.js changelogs to see if the plugin needs to be updated or not (the CSS support for example, is extremely hacky, and could break at any canary release if they decide to change the shape of the Webpack rules)

**Comment by martpie** [comment]
tl,dr: `next-transpile-modiles` hacks the internal Next.js webpack configuration to bypass the limitation of not being able to transpile code from `node_modules`. Even a minor change to a include's regex or a rule option change from the Next.js team can break `next-transpile-modules` without any warning.

**Comment by peduarte** [comment]
Im having the same issues after upgrading to Next 9.4 😭

**Comment by jacobrask** [comment]
In our Yarn monorepo we have one app using next-transpile-modules 2.3.1 and that still works, but 3.0.0 in a next project fails, in case that helps with finding the issue, or people could downgrade while the issue is being resolved in 3.x.x.

**Comment by martpie** [comment]
Not the first time I hear that, thanks for sharing 🤔 

I unfortunately still did not have time to look into the issue. 

I personally did not have trouble in various Next.js 9.4 setups (with Yarn Workspaces), so for all the people landing on this issue, please share your setups specifities so we can try to identify the root of the problem more easily 👍 (the best of the best being a PR with a test failing)

**Comment by tbergq** [comment]
I tried downgrading to `2.3.1`, but I still get the error. I also removed flow and tested with `100_000` and got the error

```
error - ../../node_modules/test-components/src/Heading.js:12:16
Syntax error: Support for the experimental syntax 'numericSeparator' isn't currently enabled (12:17)
```

**Comment by jacobrask** [comment]
> I tried downgrading to `2.3.1`, but I still get the error. I also removed flow and tested with `100_000` and got the error
> 
> ```
> error - ../../node_modules/test-components/src/Heading.js:12:16
> Syntax error: Support for the experimental syntax 'numericSeparator' isn't currently enabled (12:17)
> ```
> 
> And yes, we have the `@babel/plugin-proposal-numeric-separator` in our babel setup

This is a separate issue I’ve seen that has been resolved by removing all node_modules folders recursively in the monorepo, undoing changes to Yarn.lock and then running yarn again. Frontend Tooling ❤️😢

**Comment by Izhaki** [comment]
Next 9.4 implements [its own alias system](https://nextjs.org/blog/next-9-4#absolute-imports-and-aliases). 

There has been changes to the webpack config that I suspect do not marry well with this plugin.

We use this plugin so to import files from other packages in a monorepo, I'm pretty sure there is no longer need for this plugin with 9.4 (for us at least).

We have this:
```js
    [
      withTM,
      {
        transpileModules: [
          'packages/core',
          'packages/axl',
          'packages/status-checker',
          'packages/the-hub'
        ]
      }
    ],
```

Here is the resultant webpack config, I'm pretty sure you'll see a few issues in it:

```json
{
  "externals": [
    "next"
  ],
  "optimization": {
    "checkWasmTypes": false,
    "nodeEnv": false,
    "splitChunks": {
      "cacheGroups": {
        "default": false,
        "vendors": false,
        "styles": {
          "name": "styles",
          "test": {},
          "chunks": "all",
          "enforce": true
        }
      }
    },
    "runtimeChunk": {
      "name": "static/runtime/webpack.js"
    },
    "minimize": false,
    "minimizer": [
      {
        "options": {
          "test": {},
          "extractComments": false,
          "cache": "/Users/izhakroe/Development/tools/packages/axl/.next/cache/next-minifier",
          "parallel": 7,
          "terserOptions": {
            "parse": {
              "ecma": 8
            },
            "compress": {
              "ecma": 5,
              "warnings": false,
              "comparisons": false,
              "inline": 2
            },
            "mangle": {
              "safari10": true
            },
            "output": {
              "ecma": 5,
              "safari10": true,
              "comments": false,
              "ascii_only": true
            }
          }
        }
      },
      {
        "__next_css_remove": true,
        "options": {
          "postcssOptions": {
            "map": {
              "inline": false,
              "annotation": false
            }
          }
        }
      }
    ]
  },
  "context": "/Users/izhakroe/Development/tools/packages/axl",
  "node": {
    "setImmediate": false
  },
  "output": {
    "path": "/Users/izhakroe/Development/tools/packages/axl/.next",
    "libraryTarget": "var",
    "hotUpdateChunkFilename": "static/webpack/[id].[hash].hot-update.js",
    "hotUpdateMainFilename": "static/webpack/[hash].hot-update.json",
    "chunkFilename": "static/chunks/[name].js",
    "strictModuleExceptionHandling": true,
    "futureEmitAssets": false,
    "webassemblyModuleFilename": "static/wasm/[modulehash].wasm"
  },
  "performance": false,
  "resolve": {
    "extensions": [
      ".mjs",
      ".js",
      ".tsx",
      ".ts",
      ".jsx",
      ".json",
      ".wasm"
    ],
    "modules": [
      "node_modules"
    ],
    "alias": {
      "next/head": "next/dist/next-server/lib/head.js",
      "next/router": "next/dist/client/router.js",
      "next/config": "next/dist/next-server/lib/runtime-config.js",
      "next/dynamic": "next/dist/next-server/lib/dynamic.js",
      "next": "/Users/izhakroe/Development/tools/node_modules/next",
      "private-next-pages": "/Users/izhakroe/Development/tools/packages/axl/pages",
      "private-dot-next": "/Users/izhakroe/Development/tools/packages/axl/.next",
      "unfetch$": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/fetch/index.js",
      "isomorphic-unfetch$": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/fetch/index.js",
      "whatwg-fetch$": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/fetch/whatwg-fetch.js",
      "object-assign$": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/object-assign.js",
      "object.assign/auto": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/object.assign/auto.js",
      "object.assign/implementation": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/object.assign/implementation.js",
      "object.assign$": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/object.assign/index.js",
      "object.assign/polyfill": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/object.assign/polyfill.js",
      "object.assign/shim": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/object.assign/shim.js",
      "url": "/Users/izhakroe/Development/tools/node_modules/native-url/dist/index.js",
      "core": "/Users/izhakroe/Development/tools/packages/core/src",
      "axl": "/Users/izhakroe/Development/tools/packages/axl/src",
      "status-checker": "/Users/izhakroe/Development/tools/packages/status-checker/src",
      "the-hub": "/Users/izhakroe/Development/tools/packages/the-hub/src"
    },
    "mainFields": [
      "browser",
      "module",
      "main"
    ],
    "plugins": [
      {
        "topLevelLoader": {}
      }
    ]
  },
  "resolveLoader": {
    "alias": {
      "emit-file-loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/webpack/loaders/emit-file-loader",
      "error-loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/webpack/loaders/error-loader",
      "next-babel-loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/webpack/loaders/next-babel-loader",
      "next-client-pages-loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/webpack/loaders/next-client-pages-loader",
      "next-data-loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/webpack/loaders/next-data-loader",
      "next-serverless-loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/webpack/loaders/next-serverless-loader",
      "noop-loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/webpack/loaders/noop-loader",
      "next-plugin-loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/webpack/loaders/next-plugin-loader"
    },
    "modules": [
      "node_modules"
    ],
    "plugins": [
      {
        "topLevelLoader": {}
      }
    ]
  },
  "module": {
    "rules": [
      {
        "test": {},
        "include": [
          "/Users/izhakroe/Development/tools/packages/axl",
          {},
          {},
          {},
          {}
        ],
        "use": [
          "/Users/izhakroe/Development/tools/node_modules/next/node_modules/@next/react-refresh-utils/loader.js",
          {
            "loader": "next-babel-loader",
            "options": {
              "isServer": false,
              "distDir": "/Users/izhakroe/Development/tools/packages/axl/.next",
              "pagesDir": "/Users/izhakroe/Development/tools/packages/axl/pages",
              "cwd": "/Users/izhakroe/Development/tools/packages/axl",
              "cache": true,
              "babelPresetPlugins": [],
              "hasModern": false,
              "development": true,
              "hasReactRefresh": true
            }
          }
        ]
      },
      {
        "oneOf": [
          {
            "test": {},
            "loader": "noop-loader",
            "options": {
              "__next_css_remove": true
            }
          },
          {
            "test": {},
            "issuer": {
              "test": {}
            },
            "use": {
              "loader": "error-loader",
              "options": {
                "reason": "CSS \u001b[1mcannot\u001b[22m be imported within \u001b[36mpages/_document.js\u001b[39m. Please move global styles to \u001b[36mpages/_app.js\u001b[39m."
              }
            }
          },
          {
            "sideEffects": false,
            "test": {},
            "issuer": {
              "include": [
                "/Users/izhakroe/Development/tools/packages/axl"
              ],
              "exclude": {}
            },
            "use": [
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/node_modules/style-loader/dist/cjs.js",
                "options": {}
              },
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/node_modules/css-loader/dist/cjs.js",
                "options": {
                  "importLoaders": 1,
                  "sourceMap": true,
                  "onlyLocals": false,
                  "modules": {
                    "mode": "pure"
                  }
                }
              },
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/compiled/postcss-loader/index.js",
                "options": {
                  "ident": "__nextjs_postcss",
                  "plugins": [
                    null,
                    null
                  ],
                  "sourceMap": true
                }
              }
            ]
          },
          {
            "sideEffects": false,
            "test": {},
            "issuer": {
              "include": [
                "/Users/izhakroe/Development/tools/packages/axl"
              ],
              "exclude": {}
            },
            "use": [
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/node_modules/style-loader/dist/cjs.js",
                "options": {}
              },
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/node_modules/css-loader/dist/cjs.js",
                "options": {
                  "importLoaders": 3,
                  "sourceMap": true,
                  "onlyLocals": false,
                  "modules": {
                    "mode": "pure"
                  }
                }
              },
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/compiled/postcss-loader/index.js",
                "options": {
                  "ident": "__nextjs_postcss",
                  "plugins": [
                    null,
                    null
                  ],
                  "sourceMap": true
                }
              },
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/resolve-url-loader/index.js",
                "options": {
                  "sourceMap": true
                }
              },
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/node_modules/sass-loader/dist/cjs.js",
                "options": {
                  "sourceMap": true,
                  "sassOptions": {}
                }
              }
            ]
          },
          {
            "test": [
              {},
              {}
            ],
            "use": {
              "loader": "error-loader",
              "options": {
                "reason": "CSS Modules \u001b[1mcannot\u001b[22m be imported from within \u001b[1mnode_modules\u001b[22m.\nRead more: https://err.sh/next.js/css-modules-npm"
              }
            }
          },
          {
            "sideEffects": true,
            "test": {},
            "issuer": {
              "include": "/Users/izhakroe/Development/tools/packages/axl/pages/_app.jsx"
            },
            "use": [
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/node_modules/style-loader/dist/cjs.js",
                "options": {}
              },
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/node_modules/css-loader/dist/cjs.js",
                "options": {
                  "importLoaders": 1,
                  "sourceMap": true
                }
              },
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/compiled/postcss-loader/index.js",
                "options": {
                  "ident": "__nextjs_postcss",
                  "plugins": [
                    null,
                    null
                  ],
                  "sourceMap": true
                }
              }
            ]
          },
          {
            "sideEffects": true,
            "test": {},
            "issuer": {
              "include": "/Users/izhakroe/Development/tools/packages/axl/pages/_app.jsx"
            },
            "use": [
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/node_modules/style-loader/dist/cjs.js",
                "options": {}
              },
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/node_modules/css-loader/dist/cjs.js",
                "options": {
                  "importLoaders": 3,
                  "sourceMap": true
                }
              },
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/compiled/postcss-loader/index.js",
                "options": {
                  "ident": "__nextjs_postcss",
                  "plugins": [
                    null,
                    null
                  ],
                  "sourceMap": true
                }
              },
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/resolve-url-loader/index.js",
                "options": {
                  "sourceMap": true
                }
              },
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/node_modules/sass-loader/dist/cjs.js",
                "options": {
                  "sourceMap": true,
                  "sassOptions": {}
                }
              }
            ]
          },
          {
            "test": [
              {},
              {}
            ],
            "issuer": {
              "include": [
                {}
              ]
            },
            "use": {
              "loader": "error-loader",
              "options": {
                "reason": "Global CSS \u001b[1mcannot\u001b[22m be imported from within \u001b[1mnode_modules\u001b[22m.\nRead more: https://err.sh/next.js/css-npm"
              }
            }
          },
          {
            "test": [
              {},
              {}
            ],
            "use": {
              "loader": "error-loader",
              "options": {
                "reason": "Global CSS \u001b[1mcannot\u001b[22m be imported from files other than your \u001b[1mCustom <App>\u001b[22m. Please move all global CSS imports to \u001b[36mpages/_app.jsx\u001b[39m.\nRead more: https://err.sh/next.js/css-global"
              }
            }
          },
          {
            "issuer": {
              "test": {}
            },
            "exclude": [
              {},
              {},
              {}
            ],
            "use": {
              "loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/compiled/file-loader/cjs.js",
              "options": {
                "name": "static/media/[name].[hash].[ext]"
              }
            }
          }
        ]
      },
      {
        "test": {},
        "use": [
          "extracted-loader",
          "/Users/izhakroe/Development/tools/node_modules/@zeit/next-css/node_modules/mini-css-extract-plugin/dist/loader.js",
          {
            "loader": "css-loader",
            "options": {
              "modules": true,
              "minimize": false,
              "sourceMap": true,
              "importLoaders": 0,
              "camelCase": true
            }
          }
        ]
      },
      {
        "test": {},
        "use": [
          "extracted-loader",
          "/Users/izhakroe/Development/tools/node_modules/@zeit/next-css/node_modules/mini-css-extract-plugin/dist/loader.js",
          {
            "loader": "css-loader",
            "options": {
              "modules": true,
              "minimize": false,
              "sourceMap": true,
              "importLoaders": 1,
              "camelCase": true
            }
          },
          {
            "loader": "sass-loader",
            "options": {}
          }
        ]
      },
      {
        "test": {},
        "use": [
          "extracted-loader",
          "/Users/izhakroe/Development/tools/node_modules/@zeit/next-css/node_modules/mini-css-extract-plugin/dist/loader.js",
          {
            "loader": "css-loader",
            "options": {
              "modules": true,
              "minimize": false,
              "sourceMap": true,
              "importLoaders": 1,
              "camelCase": true
            }
          },
          {
            "loader": "sass-loader",
            "options": {}
          }
        ]
      },
      {
        "test": {},
        "loader": "eslint-loader",
        "enforce": "pre",
        "options": {
          "emitWarning": true,
          "ignorePath": "../../.eslintignore"
        }
      }
    ],
    "strictExportPresence": true
  },
  "plugins": [
    {},
    {},
    {
      "definitions": {
        "process.env.MONGO_SERVICE": "\"https://axl-mongo.qa.devops.csr.pstars\"",
        "process.env.MONGO_CDN": "\"https://thehub.rationalcdn.com/axlmongo/qa\"",
        "process.env.TWITCH_SERVICE": "\"https://twitchhub.qa.devops.csr.pstars/live\"",
        "process.env.TWITCH_ADMIN": "\"https://twitch-poller.qa.devops.csr.pstars/twitch\"",
        "process.env.VIDEOS_SERVICE": "\"https://myvideos.qa.devops.csr.pstars\"",
        "process.env.FEED_SERVICE": "\"https://personalfeed.qa.devops.csr.pstars\"",
        "process.env.AXL_OLAP": "\"https://axl-olap.qa.devops.csr.pstars\"",
        "process.env.BUILD_NUMBER": "\"N/A\"",
        "process.env.AKAMAI_CDN_API_KEY": "\"ghKC6p3sd5E5BhjC030bNm4965DqMevztVHXdXhLkUo9\"",
        "process.env.AKAMAI_CDN_UPLOAD_HOST": "\"pkrdeo-nsu.akamaihd.net\"",
        "process.env.AKAMAI_CDN_ROOT_PATH": "\"/952772/thehub/axlmongo/dev/\"",
        "process.env.AKAMAI_CDN_KEY_NAME": "\"thehub\"",
        "process.env.AKAMAI_CDN_PUBLIC_HOST": "\"thehub.rationalcdn.com\"",
        "process.env.AKAMAI_CDN_PUBLIC_PATH": "\"/axlmongo/dev/\"",
        "process.env.ASSETS_PREFIX": "\"\"",
        "process.env.NODE_ENV": "\"development\"",
        "process.browser": "true",
        "process.env.__NEXT_DIST_DIR": "\"/Users/izhakroe/Development/tools/packages/axl/.next\"",
        "process.env.__NEXT_EXPORT_TRAILING_SLASH": "false",
        "process.env.__NEXT_MODERN_BUILD": "false",
        "process.env.__NEXT_GRANULAR_CHUNKS": "false",
        "process.env.__NEXT_BUILD_INDICATOR": "true",
        "process.env.__NEXT_PRERENDER_INDICATOR": "false",
        "process.env.__NEXT_PLUGINS": "false",
        "process.env.__NEXT_STRICT_MODE": "false",
        "process.env.__NEXT_REACT_MODE": "\"legacy\"",
        "process.env.__NEXT_ROUTER_BASEPATH": "\"\""
      }
    },
    {
      "filename": "react-loadable-manifest.json"
    },
    {
      "ampPages": {}
    },
    {
      "prevAssets": {}
    },
    {},
    {
      "prevAssets": null
    },
    {
      "_originalSettings": {
        "filename": "[name]_[hash].js",
        "path": "./static/development/dll",
        "context": "/Users/izhakroe/Development/tools/packages/axl",
        "entry": {
          "dll": [
            "react",
            "react-dom"
          ]
        },
        "config": {
          "devtool": "cheap-module-source-map",
          "mode": "development",
          "resolve": {
            "extensions": [
              ".mjs",
              ".js",
              ".tsx",
              ".ts",
              ".jsx",
              ".json",
              ".wasm"
            ],
            "modules": [
              "node_modules"
            ],
            "alias": {
              "next/head": "next/dist/next-server/lib/head.js",
              "next/router": "next/dist/client/router.js",
              "next/config": "next/dist/next-server/lib/runtime-config.js",
              "next/dynamic": "next/dist/next-server/lib/dynamic.js",
              "next": "/Users/izhakroe/Development/tools/node_modules/next",
              "private-next-pages": "/Users/izhakroe/Development/tools/packages/axl/pages",
              "private-dot-next": "/Users/izhakroe/Development/tools/packages/axl/.next",
              "unfetch$": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/fetch/index.js",
              "isomorphic-unfetch$": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/fetch/index.js",
              "whatwg-fetch$": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/fetch/whatwg-fetch.js",
              "object-assign$": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/object-assign.js",
              "object.assign/auto": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/object.assign/auto.js",
              "object.assign/implementation": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/object.assign/implementation.js",
              "object.assign$": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/object.assign/index.js",
              "object.assign/polyfill": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/object.assign/polyfill.js",
              "object.assign/shim": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/polyfills/object.assign/shim.js",
              "url": "/Users/izhakroe/Development/tools/node_modules/native-url/dist/index.js",
              "core": "/Users/izhakroe/Development/tools/packages/core/src",
              "axl": "/Users/izhakroe/Development/tools/packages/axl/src",
              "status-checker": "/Users/izhakroe/Development/tools/packages/status-checker/src",
              "the-hub": "/Users/izhakroe/Development/tools/packages/the-hub/src"
            },
            "mainFields": [
              "browser",
              "module",
              "main"
            ],
            "plugins": [
              {
                "topLevelLoader": {}
              }
            ]
          }
        }
      }
    },
    {
      "options": {},
      "fullBuildTimeout": 200,
      "requestTimeout": 10000
    },
    {
      "buildId": "development",
      "clientManifest": true,
      "modern": false
    },
    {},
    {
      "options": {
        "filename": "static/css/[name].css",
        "chunkFilename": "static/css/[name].chunk.css"
      }
    },
    {
      "definitions": {}
    }
  ],
  "mode": "development",
  "name": "client",
  "target": "web",
  "devtool": "eval-source-map"
}
{
  "externals": [
    null
  ],
  "optimization": {
    "checkWasmTypes": false,
    "nodeEnv": false,
    "splitChunks": false,
    "minimize": false,
    "minimizer": [
      {
        "options": {
          "test": {},
          "extractComments": false,
          "cache": "/Users/izhakroe/Development/tools/packages/axl/.next/cache/next-minifier",
          "parallel": 7,
          "terserOptions": {
            "parse": {
              "ecma": 8
            },
            "compress": {
              "ecma": 5,
              "warnings": false,
              "comparisons": false,
              "inline": 2
            },
            "mangle": {
              "safari10": true
            },
            "output": {
              "ecma": 5,
              "safari10": true,
              "comments": false,
              "ascii_only": true
            }
          }
        }
      },
      {
        "__next_css_remove": true,
        "options": {
          "postcssOptions": {
            "map": {
              "inline": false,
              "annotation": false
            }
          }
        }
      }
    ]
  },
  "context": "/Users/izhakroe/Development/tools/packages/axl",
  "node": {
    "setImmediate": false
  },
  "output": {
    "path": "/Users/izhakroe/Development/tools/packages/axl/.next/server",
    "libraryTarget": "commonjs2",
    "hotUpdateChunkFilename": "static/webpack/[id].[hash].hot-update.js",
    "hotUpdateMainFilename": "static/webpack/[hash].hot-update.json",
    "chunkFilename": "[name].js",
    "strictModuleExceptionHandling": true,
    "futureEmitAssets": false,
    "webassemblyModuleFilename": "static/wasm/[modulehash].wasm"
  },
  "performance": false,
  "resolve": {
    "extensions": [
      ".js",
      ".mjs",
      ".tsx",
      ".ts",
      ".jsx",
      ".json",
      ".wasm"
    ],
    "modules": [
      "node_modules"
    ],
    "alias": {
      "next/head": "next/dist/next-server/lib/head.js",
      "next/router": "next/dist/client/router.js",
      "next/config": "next/dist/next-server/lib/runtime-config.js",
      "next/dynamic": "next/dist/next-server/lib/dynamic.js",
      "next": "/Users/izhakroe/Development/tools/node_modules/next",
      "private-next-pages": "/Users/izhakroe/Development/tools/packages/axl/pages",
      "private-dot-next": "/Users/izhakroe/Development/tools/packages/axl/.next",
      "core": "/Users/izhakroe/Development/tools/packages/core/src",
      "axl": "/Users/izhakroe/Development/tools/packages/axl/src",
      "status-checker": "/Users/izhakroe/Development/tools/packages/status-checker/src",
      "the-hub": "/Users/izhakroe/Development/tools/packages/the-hub/src"
    },
    "mainFields": [
      "main",
      "module"
    ],
    "plugins": [
      {
        "topLevelLoader": {}
      }
    ]
  },
  "resolveLoader": {
    "alias": {
      "emit-file-loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/webpack/loaders/emit-file-loader",
      "error-loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/webpack/loaders/error-loader",
      "next-babel-loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/webpack/loaders/next-babel-loader",
      "next-client-pages-loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/webpack/loaders/next-client-pages-loader",
      "next-data-loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/webpack/loaders/next-data-loader",
      "next-serverless-loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/webpack/loaders/next-serverless-loader",
      "noop-loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/webpack/loaders/noop-loader",
      "next-plugin-loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/build/webpack/loaders/next-plugin-loader"
    },
    "modules": [
      "node_modules"
    ],
    "plugins": [
      {
        "topLevelLoader": {}
      }
    ]
  },
  "module": {
    "rules": [
      {
        "test": {},
        "include": [
          "/Users/izhakroe/Development/tools/packages/axl",
          {},
          {},
          {},
          {}
        ],
        "use": {
          "loader": "next-babel-loader",
          "options": {
            "isServer": true,
            "distDir": "/Users/izhakroe/Development/tools/packages/axl/.next",
            "pagesDir": "/Users/izhakroe/Development/tools/packages/axl/pages",
            "cwd": "/Users/izhakroe/Development/tools/packages/axl",
            "cache": true,
            "babelPresetPlugins": [],
            "hasModern": false,
            "development": true,
            "hasReactRefresh": false
          }
        }
      },
      {
        "oneOf": [
          {
            "test": {},
            "loader": "noop-loader",
            "options": {
              "__next_css_remove": true
            }
          },
          {
            "test": {},
            "issuer": {
              "test": {}
            },
            "use": {
              "loader": "error-loader",
              "options": {
                "reason": "CSS \u001b[1mcannot\u001b[22m be imported within \u001b[36mpages/_document.js\u001b[39m. Please move global styles to \u001b[36mpages/_app.js\u001b[39m."
              }
            }
          },
          {
            "sideEffects": false,
            "test": {},
            "issuer": {
              "include": [
                "/Users/izhakroe/Development/tools/packages/axl"
              ],
              "exclude": {}
            },
            "use": [
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/node_modules/css-loader/dist/cjs.js",
                "options": {
                  "importLoaders": 1,
                  "sourceMap": true,
                  "onlyLocals": true,
                  "modules": {
                    "mode": "pure"
                  }
                }
              },
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/compiled/postcss-loader/index.js",
                "options": {
                  "ident": "__nextjs_postcss",
                  "plugins": [
                    null,
                    null
                  ],
                  "sourceMap": true
                }
              }
            ]
          },
          {
            "sideEffects": false,
            "test": {},
            "issuer": {
              "include": [
                "/Users/izhakroe/Development/tools/packages/axl"
              ],
              "exclude": {}
            },
            "use": [
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/node_modules/css-loader/dist/cjs.js",
                "options": {
                  "importLoaders": 3,
                  "sourceMap": true,
                  "onlyLocals": true,
                  "modules": {
                    "mode": "pure"
                  }
                }
              },
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/dist/compiled/postcss-loader/index.js",
                "options": {
                  "ident": "__nextjs_postcss",
                  "plugins": [
                    null,
                    null
                  ],
                  "sourceMap": true
                }
              },
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/resolve-url-loader/index.js",
                "options": {
                  "sourceMap": true
                }
              },
              {
                "loader": "/Users/izhakroe/Development/tools/node_modules/next/node_modules/sass-loader/dist/cjs.js",
                "options": {
                  "sourceMap": true,
                  "sassOptions": {}
                }
              }
            ]
          },
          {
            "test": [
              {},
              {}
            ],
            "use": {
              "loader": "error-loader",
              "options": {
                "reason": "CSS Modules \u001b[1mcannot\u001b[22m be imported from within \u001b[1mnode_modules\u001b[22m.\nRead more: https://err.sh/next.js/css-modules-npm"
              }
            }
          },
          {
            "test": [
              {},
              {}
            ],
            "use": "/Users/izhakroe/Development/tools/node_modules/next/dist/compiled/ignore-loader/index.js"
          },
          {
            "test": [
              {},
              {}
            ],
            "issuer": {
              "include": [
                {}
              ]
            },
            "use": {
              "loader": "error-loader",
              "options": {
                "reason": "Global CSS \u001b[1mcannot\u001b[22m be imported from within \u001b[1mnode_modules\u001b[22m.\nRead more: https://err.sh/next.js/css-npm"
              }
            }
          },
          {
            "test": [
              {},
              {}
            ],
            "use": {
              "loader": "error-loader",
              "options": {
                "reason": "Global CSS \u001b[1mcannot\u001b[22m be imported from files other than your \u001b[1mCustom <App>\u001b[22m. Please move all global CSS imports to \u001b[36mpages/_app.jsx\u001b[39m.\nRead more: https://err.sh/next.js/css-global"
              }
            }
          }
        ]
      },
      {
        "test": {},
        "use": [
          {
            "loader": "css-loader/locals",
            "options": {
              "modules": true,
              "minimize": false,
              "sourceMap": true,
              "importLoaders": 0,
              "camelCase": true
            }
          }
        ]
      },
      {
        "test": {},
        "use": [
          {
            "loader": "css-loader/locals",
            "options": {
              "modules": true,
              "minimize": false,
              "sourceMap": true,
              "importLoaders": 1,
              "camelCase": true
            }
          },
          {
            "loader": "sass-loader",
            "options": {}
          }
        ]
      },
      {
        "test": {},
        "use": [
          {
            "loader": "css-loader/locals",
            "options": {
              "modules": true,
              "minimize": false,
              "sourceMap": true,
              "importLoaders": 1,
              "camelCase": true
            }
          },
          {
            "loader": "sass-loader",
            "options": {}
          }
        ]
      },
      {
        "test": {},
        "loader": "eslint-loader",
        "enforce": "pre",
        "options": {
          "emitWarning": true,
          "ignorePath": "../../.eslintignore"
        }
      }
    ],
    "strictExportPresence": true
  },
  "plugins": [
    {},
    {
      "definitions": {
        "process.env.MONGO_SERVICE": "\"https://axl-mongo.qa.devops.csr.pstars\"",
        "process.env.MONGO_CDN": "\"https://thehub.rationalcdn.com/axlmongo/qa\"",
        "process.env.TWITCH_SERVICE": "\"https://twitchhub.qa.devops.csr.pstars/live\"",
        "process.env.TWITCH_ADMIN": "\"https://twitch-poller.qa.devops.csr.pstars/twitch\"",
        "process.env.VIDEOS_SERVICE": "\"https://myvideos.qa.devops.csr.pstars\"",
        "process.env.FEED_SERVICE": "\"https://personalfeed.qa.devops.csr.pstars\"",
        "process.env.AXL_OLAP": "\"https://axl-olap.qa.devops.csr.pstars\"",
        "process.env.BUILD_NUMBER": "\"N/A\"",
        "process.env.AKAMAI_CDN_API_KEY": "\"ghKC6p3sd5E5BhjC030bNm4965DqMevztVHXdXhLkUo9\"",
        "process.env.AKAMAI_CDN_UPLOAD_HOST": "\"pkrdeo-nsu.akamaihd.net\"",
        "process.env.AKAMAI_CDN_ROOT_PATH": "\"/952772/thehub/axlmongo/dev/\"",
        "process.env.AKAMAI_CDN_KEY_NAME": "\"thehub\"",
        "process.env.AKAMAI_CDN_PUBLIC_HOST": "\"thehub.rationalcdn.com\"",
        "process.env.AKAMAI_CDN_PUBLIC_PATH": "\"/axlmongo/dev/\"",
        "process.env.ASSETS_PREFIX": "\"\"",
        "process.env.NODE_ENV": "\"development\"",
        "process.browser": "false",
        "process.env.__NEXT_EXPORT_TRAILING_SLASH": "false",
        "process.env.__NEXT_MODERN_BUILD": "false",
        "process.env.__NEXT_GRANULAR_CHUNKS": "false",
        "process.env.__NEXT_BUILD_INDICATOR": "true",
        "process.env.__NEXT_PRERENDER_INDICATOR": "false",
        "process.env.__NEXT_PLUGINS": "false",
        "process.env.__NEXT_STRICT_MODE": "false",
        "process.env.__NEXT_REACT_MODE": "\"legacy\"",
        "process.env.__NEXT_ROUTER_BASEPATH": "\"\"",
        "global.GENTLY": "false"
      }
    },
    {
      "prevAssets": {}
    },
    {},
    {
      "prevAssets": null
    },
    {
      "serverless": false
    },
    {
      "options": {
        "outputPath": "/Users/izhakroe/Development/tools/packages/axl/.next/server"
      }
    },
    {},
    {},
    {
      "definitions": {}
    }
  ],
  "mode": "development",
  "name": "server",
  "target": "node",
  "devtool": "eval-source-map"
}
```

**Comment by TrySound** [comment]
Use spoiler please

**Comment by tbergq** [comment]
Actually I found the issue, it seems like the issue is with `next-plugin-custom-babel-config`, not `next-transpile-modules`. I noticed that rules are now

```
{
  test: /\.(tsx|ts|js|mjs|jsx)$/,
  include: [
    '/Users/trondbergquist/code/privat/next-9.4-test/src/app',
    /next[\\/]dist[\\/]next-server[\\/]lib/,
    /next[\\/]dist[\\/]client/,
    /next[\\/]dist[\\/]pages/,
    /[\\/](strip-ansi|ansi-regex)[\\/]/
  ],
  exclude: [Function: exclude],
  use: [
    '/Users/trondbergquist/code/privat/next-9.4-test/node_modules/@next/react-refresh-utils/loader.js',
    { loader: 'next-babel-loader', options: [Object] }
  ]
}
```
and `next-plugin-custom-babel-config` tries to do ` rule.use.options.configFile = nextConfig.babelConfigFile;` which is probably why it works for production builds, since there is no `react-fast-refresh` there

**Comment by tbergq** [comment]
I can confirm it works now with `next-plugin-custom-babel-config@1.0.3`, so feel free to close the issue @martpie

**Comment by martpie** [comment]
Great you managed to find the root of the issue! I'll be waiting on the confirmation of others and then we can close it 👍

**Comment by Izhaki** [comment]
We don't have `next-plugin-custom-babel-config` anywhere in our project. It's a dependency of which package exactly?

```
    "next": "9.4.2",
    "next-compose-plugins": "^2.2.0",
    "next-transpile-modules": "^3.3.0",
```

**Comment by tbergq** [comment]
> We don't have `next-plugin-custom-babel-config` anywhere in our project. It's a dependency of which package exactly?
> 
> ```
>     "next": "9.4.2",
>     "next-compose-plugins": "^2.2.0",
>     "next-transpile-modules": "^3.3.0",
> ```

It is its own [package](https://github.com/josephluck/next-plugin-custom-babel-config#i-need-to-transpile-my-workspaces)

**Comment by Izhaki** [comment]
Well in that case we don't use it and plugin is broken for us with 9.4

**Comment by Izhaki** [comment]
OK. I removed this plugin from our repo and I'm getting exactly the same error is with it.

So clearly the issue is not with `next-transpile-modules` for us.

**Comment by martpie** [comment]
Thank you for the feedback guys 👍 

If anyone is passing by and the above "solutions" did not fix it for you, please comment :)

**Comment by nnennajohn** [comment]
Hi, 

I am still having this issue. The comment above pointed out its a plugin by nextjs that might be causing it. But I don't see a solution. Was anyone able to get working around this? I am on yarn berry, next 9.45 and next-transpile-modules 3.3.0. 

Everything worked before the next upgrade, and I don't want to downgrade.

<img width="884" alt="Screen Shot 2020-06-09 at 2 27 19 AM" src="https://user-images.githubusercontent.com/1761197/84131120-0095f580-a9f9-11ea-859a-b0abea34bf4a.png">

**Comment by RuniBr** [comment]
@andrewmclagan Am using Yarn 2 with Plug'n'Play and after banging my head against the wall for a couple weeks, was finally able to make some progress.

Firstly, I needed to change the workspace names passed to `transpile-next-modules` to absolute paths instead. I.e. `require('next-transpile-modules')(['@app/lib-next-utils']);` would become `require('next-transpile-modules')(['/Users/runib/work/app/packages/libs/next-utils']);`.

Secondly, listing `peerDependencies` in `@app/lib-next-utils` breaks transpilation. If I omit all peer dependencies this plugin correctly transpiles the `@app/lib-next-utils` workspace; adding any peer dependency back in and transpilation breaks again. Removing all peer dependencies isn't a solution though, at least not an appropriate one in my particular case. Have a feeling it's related to how `PnP`  works with dependencies/transitive dependencies

**Comment by nnennajohn** [comment]
@RuniBr Thanks for the info. Unfortunately, I also run this in docker, so can't be hard-coding paths. 

I think something changed in the latest nextjs, and this package needs to update to work with it. I got it working by following the hint above, and using next-plugin-transpile-modules instead.

<img width="996" alt="Screen Shot 2020-06-09 at 6 18 23 PM" src="https://user-images.githubusercontent.com/1761197/84216377-2b765d00-aa7e-11ea-97ee-d96be09ccef4.png">

**Comment by andrewmclagan** [comment]
I believe regarding the support for yarn v2 or yarn plug-and-play in general should be done in a seperate issue, not this issue. @nnennajohn @RuniBr

**Comment by nnennajohn** [comment]
@andrewmclagan I mentioned it because i'd used yarn v2 since it launched, and this package always worked with it. It only broke with the recent next update, so I don't think it's a yarn issue. Switching to next-plugin-transpile-modules got it working again.

**Comment by martpie** [comment]
@nnennajohn `next-plugin-transpile-modules` is just the old package name of `next-transpile-modules`, I would recommend you use `next-transpile-modules@2` if you don't need to use the latest feature of v3 (CSS module, `.mjs` etc), it has a lot of bugfixes that were not released under `next-plugin-transpile-modules`

**Comment by martpie** [comment]
To anyone else snooping around: I think the original problem of this issue was fixed and looks unrelated to the problem you are facing, if you'd mind opening a separate issue, with clear reproduction steps, I could spend some time debugging it :)

**Comment by nnennajohn** [comment]
@martpie that's interesting. Because I thought I was on the latest. Will change it tonight and report back if I am able to get it working. Thank you.

**Comment by jamieowen** [comment]
Hey, not sure on status of this.. but i had the same issue.

Fixed by passing resolved absolute paths instead of package names, as mentioned above.
( on a very minimal config - next 9.4.0, yarn 1.22.4, next-transpile-modules 3.3.0  )
```
const path = require('path');

const withTM = require('next-transpile-modules')(
  path.resolve( __dirname,'../../packages/ui/src'),
  path.resolve(__dirname,'../../packages/ui-rendering/src')
);

module.exports = withTM();
```

I also declared paths in tsconfig.json using the same with next alias support. ( no babel config )
( i still seemed to need the next-transpile-modules plugin )

**Comment by martpie** [comment]
Closing this in favor of #101 as it seems to be the root issue, and a workaround exists

**Comment by softmarshmallow** [comment]
I changed `.babelrc` to `babel.config.js` and it worked fine. :) - as it was mentioned on README. 
Think it should be with a warning or something. It's easy to be missed.
